RedditSharp/Interfaces/ISettableWebAgent.cs

```diff
@@ -0,0 +1,10 @@
+using Newtonsoft.Json;
+
+namespace RedditSharp.Interfaces
+{
+    internal interface ISettableWebAgent
```

Comment:
What's the purpose behind having this as an interface here?

Reply:
It's for [PopulateObjects](https://github.com/CrustyJew/RedditSharp/pull/201/files#diff-6454e46a967ed7c5dee9b5a074712b3aR23) to have a way to set the web agent for each object. So when you fetch the collections, you can do collection operations (e.g. `AddPostAsync`) directly to the object instead of passing in the web agent, for example. 
Do you prefer the web agents to be set after populating the objects [here](https://github.com/CrustyJew/RedditSharp/pull/201/files#diff-031fc70246efa6dcd09d601a17e0fd99R960):
```C#
var result = Helpers.PopulateObjects<Collection>(json, WebAgent);
foreach (var collection in result)
    collection.WebAgent = WebAgent;
return result;
```

Or pass in `Action<T>` (e.g. `x => x.WebAgent = WebAgent`) that gets called after the item is populated?

Or something else?

Reply:
I'm thinking it may make more sense just to make the WebAgent settable by default instead of having a specific interface for it.